src/ocean/time/StopWatch.d

```diff
@@ -96,11 +107,11 @@ public struct StopWatch
 
         private static ulong timer ()
         {
-                timeval tv;
-                if (gettimeofday (&tv, null))
-                    throw new PlatformException ("Timer :: linux timer is not available");
+                timespec t;
+                if (clock_gettime(CLOCK_MONOTONIC, &t))
+                    throw new PlatformException ("Timer :: CLOCK_MONOTONIC is not available");
 
-                return (cast(ulong) tv.tv_sec * 1_000_000) + tv.tv_usec;
+                return t.tv_sec * 1_000_000_000UL + t.tv_nsec / 1_000UL;
```

Comment:
Should this be `tv_sec * 1_000_000 + t.tv_nsec / 1_000` ? Looks like you're adding nanoseconds (from the first member) and microseconds (from the second member).